exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -159,96 +162,153 @@ def collect(self) -> None:
+                            "bucket_counts": number_data_point.bucket_counts,
+                            "explicit_bounds": (
+                                number_data_point.explicit_bounds
+                            ),
+                            "sum": number_data_point.sum,
+                        }
+                    )
+                else:
+                    values.append(number_data_point.value)
+
+            metric_name = ""
+            if self._prefix != "":
+                metric_name = self._prefix + "_"
+            metric_name += self._sanitize(metric.name)
+
+            description = metric.description or ""
+
+            pre_metric_family_ids = []
+
+            for label_keys in label_keyss:
+                pre_metric_family_ids.append(
+                    "|".join(
+                        [
+                            metric_name,
+                            description,
+                            "%".join(label_keys),
+                            metric.unit,
+                        ]
+                    )
                 )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values, value=metric.point.value
-            )
-        elif isinstance(metric.point, Gauge):
-
-            metric_family_id = "|".join(
-                [metric_family_id, GaugeMetricFamily.__name__]
-            )
-
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = GaugeMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
-                )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values, value=metric.point.value
-            )
-        elif isinstance(metric.point, Histogram):
-
-            metric_family_id = "|".join(
-                [metric_family_id, HistogramMetricFamily.__name__]
-            )
 
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = HistogramMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
-                )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values,
-                buckets=_convert_buckets(metric),
-                sum_value=metric.point.sum,
-            )
-        else:
-            _logger.warning("Unsupported metric type. %s", type(metric.point))
+            for pre_metric_family_id, label_values, value in zip(
+                pre_metric_family_ids, label_valuess, values
+            ):
```

Comment:
IMO the code would be easier to follow if we did this in one pass instead. But I don't want to block this PR and we can remove a bunch of this logic once "short name" is added

Reply:
Same as above :+1: